docs/docs/adding-forms.md

```diff
@@ -2,7 +2,169 @@
+When a user types into an input box, the state should update. Add an `onChange` prop to update state and add a `value` prop to keep the input up to date with the new state:
+
+```JSX
+import React from "react"
+
+export default class IndexPage extends React.Component {
+    state = {
+        firstName: '',
+        lastName: '',
+    }
+
+    handleInputChange = (event) => {
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+
+        this.setState({
+            [name]: value,
+        });
+    }
+
+    render() {
+        return (
+            <form>
+                <input
+                    type="text"
+                    name="firstName"
+                    value={this.state.firstName}
+                    onChange={this.handleInputChange}
+                />
+                <input
+                    type="text"
+                    name="lastName"
+                    value={this.state.lastName}
+                    onChange={this.handleInputChange}
+                />
+                <input
+                    type="submit"
+                    value="Submit"
+                />
+            </form>
+        )
+    }
+}
+```
+
+Now that our inputs are working, we want something to happen when we submit the form. Add `onSubmit` props to the form element and add `handleSubmit` to show an alert when the user submits the form:
+
+```JSX
+import React from "react"
+
+export default class IndexPage extends React.Component {
+    state = {
+        firstName: '',
+        lastName: '',
+    }
+
+    handleInputChange = (event) => {
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+
+        this.setState({
+            [name]: value,
+        });
+    }
+
+    handleSubmit = (event) => {
+        event.preventDefault();
+        alert(`Welcome ${this.state.firstName} ${this.state.lastName}!`);
+    }
+
+    render() {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <input
+                    type="text"
+                    name="firstName"
+                    value={this.state.firstName}
+                    onChange={this.handleInputChange}
+                />
+                <input
+                    type="text"
+                    name="lastName"
+                    value={this.state.lastName}
+                    onChange={this.handleInputChange}
+                />
+                <input
+                    type="submit"
+                    value="Submit"
+                />
+            </form>
+        )
+    }
+}
+```
+
+This form isn't doing anything besides showing the user information that they just entered. At this point, you may want to move this form to a component, make a post to a server with axios, or add this submission to an array in your redux state. Gatsby is built on top of React. So anything that is possible with a React form is possible in Gatsby.
```

Comment:
Maybe we can link to some libraries here? I've really liked Formik, final-form, etc.

Reply:
Added them to the bottom!